src/Tribe/Cache_Listener.php

```diff
@@ -1,171 +1,214 @@
+	 * @param mixed  $old_value   The old option value.
+	 * @param mixed  $value       The new option value.
+	 */
+	public function update_last_save_post( $option_name, $old_value, $value ) {
+		$triggers = [
+			'tribe_events_calendar_options' => true,
+			'permalink_structure'           => true,
+			'rewrite_rules'                 => true,
+			'start_of_week'                 => true,
+		];
+
+		$triggers = $this->filter_action_last_occurrence_triggers( $triggers, static::TRIGGER_SAVE_POST, func_get_args() );
+
+		if ( ! empty( $triggers[ $option_name ] ) ) {
+			$this->cache->set_last_occurrence( self::TRIGGER_SAVE_POST );
 		}
+	}
 
-		/**
-		 * Run the caching functionality that is executed on saving tribe calendar options.
-		 *
-		 * @see 'updated_option'
-		 *
-	     * @param string $option_name    Name of the updated option.
-	     * @param mixed  $old_value The old option value.
-	     * @param mixed  $value     The new option value.
-		 */
-		public function update_last_save_post( $option_name, $old_value, $value ) {
-			$triggers = [
-				'tribe_events_calendar_options' => true,
-				'permalink_structure'           => true,
-				'rewrite_rules'                 => true,
-				'start_of_week'                 => true,
-			];
-			if ( ! empty( $triggers[ $option_name ] ) ) {
-				$this->cache->set_last_occurrence( self::TRIGGER_SAVE_POST );
-			}
+	/**
+	 * Run the caching functionality that is executed on saving tribe calendar options.
+	 *
+	 * @see   'updated_option'
+	 *
+	 * @since 4.11.0
+	 *
+	 * @param string $option_name Name of the updated option.
+	 * @param mixed  $old_value   The old option value.
+	 * @param mixed  $value       The new option value.
+	 */
+	public function update_last_updated_option( $option_name, $old_value, $value ) {
+		$triggers = [
+			'active_plugins'                => true,
+			'tribe_events_calendar_options' => true,
+			'permalink_structure'           => true,
+			'rewrite_rules'                 => true,
+			'start_of_week'                 => true,
+			'sidebars_widgets'              => true,
+			'stylesheet'                    => true,
+			'template'                      => true,
+			'WPLANG'                        => true,
+		];
+
+		$triggers = $this->filter_action_last_occurrence_triggers( $triggers, static::TRIGGER_UPDATED_OPTION, func_get_args() );
+
+		if ( ! empty( $triggers[ $option_name ] ) ) {
+			$this->cache->set_last_occurrence( self::TRIGGER_UPDATED_OPTION );
 		}
+	}
+
+	/**
+	 * Filtering for last occurrence triggers.
+	 *
+	 * @since TBD
+	 *
+	 * @param array<string,bool> $triggers Which options will trigger this given action last occurrence.
+	 * @param string             $action   Which action this trigger will set.
+	 * @param array              $args     Which arguments from the updated option method.
+	 *
+	 * @return array
+	 */
+	public function filter_action_last_occurrence_triggers( $triggers, $action, array $args = [] ) {
```

Comment:
Might be a good idea to enforce and initialize some args. 

```suggestion
	public function filter_action_last_occurrence_triggers( array $triggers = [], $action = '', array $args = [] ) {
```